app/domain/authentication/authn_k8s/application_identity.rb

```diff
@@ -84,9 +85,7 @@ def validate_constraint_combinations
         controllers = %i(deployment deployment_config stateful_set)
 
         controller_constraints = constraints.keys & controllers
```

Comment:
why do we defined controllers and constraints as 2 separate entities? It looks like it holds the same information

Reply:
and what is the `&` here for?

Reply:
controllers is a hard-coded array of the controllers that can't be combined. `constraints.keys` is the actual constraints in the application identity. 

`constraints.keys & controllers` is the intersection of the 2. So we validate that the intersection of the 2 arrays have at most one value.

Reply:
nice!